pkg/skaffold/deploy/kubectl/kubectl.go

```diff
@@ -139,6 +139,13 @@ func (k *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Art
 		"DeployerType": "kubectl",
 	})
 
+	// Check that the cluster is reachable.
+	// This gives a better error message when the cluster can't
+	// be reached.
+	if err := kubernetes.FailIfClusterIsNotReachable(); err != nil {
```

Comment:
is it possible to pass the `kubeContext` to this method, using `config.GetKubeContext()`? Otherwise this check won't happen against the `kubeContext` defined in the skaffold.yaml deploy stanza.

Reply:
hmm...I guess in theory this would be possible, but currently this calls `kubernetes.getClientSet()`, which in turn calls `kubernetes/context.GetRestClientConfig()`, and that package has the `kubeContext` variable which I guess holds the active kubecontext for that skaffold run. I'm afraid to touch that variable because my understanding is that we compute that at some point when the `RunContext` is created and it stays constant.

it feels to me like if we're setting that package-level variable correctly, then it should be respecting the `kubeContext` value from the skaffold.yaml deploy stanza right?

regardless, if we make that change I think it should happen in a separate PR.